### PR TITLE
Feature/172/172 test reports

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

- Publish Gatling reports (HTML) as downloadable pipeline artifact. (See sample at https://github.com/agera-edc/MinimumViableDataspace/actions/runs/2831636345)
- Upload test coverage data to CodeCov: https://app.codecov.io/gh/agera-edc/MinimumViableDataspace/pulls

## Why it does that

Improved reporting, aligned with EDC practices.

## Linked Issue(s)

https://github.com/agera-edc/MinimumViableDataspace/issues/172

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,9 +1,15 @@
 plugins {
     java
     `java-library`
+    jacoco
 }
 
 allprojects {
+    apply(plugin = "java")
+    if (System.getenv("JACOCO") == "true") {
```
Comment (marcgs) [inline build.gradle.kts:9]: nit: Using gradle project properties seems to be more appropriate (scope of the var is narrowed down for Gradle usage exclusively). Project properties can be passed with the `-P` option but there is a way to pass them through env vars as well: https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -57,10 +60,27 @@ jobs:
...
+
+      - name: "Publish Gatling report"
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: Gatling reports (local run)
+          path: '**/build/reports/gatling/**'
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
```
Comment (pkirch) [inline .github/workflows/cd.yaml:80]: nit: Why version 1? There is a v2 since end of July.

**Review by pkirch — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (algattik) [reply build.gradle.kts:9]: Fully agree - it seemed though easier on maintainers to follow exactly what's done in EDC

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (algattik) [reply .github/workflows/cd.yaml:80]: Same reason

**Review by cpeeyush — APPROVED** [review]

**Pull request merged**